raiden/api/python.py

```diff
@@ -487,19 +487,26 @@ def channel_batch_close(
             token_address=token_address,
             partner_addresses=partner_addresses,
         )
-        token_network_identifier = views.get_token_network_identifier_by_token_address(
-            chain_state=views.state_from_raiden(self.raiden),
-            payment_network_id=registry_address,
-            token_address=token_address,
-        )
 
+        # Skip the WAL and produce the ContractSendChannelClose events
+        # manually.
+        #
+        # This will make the closing action *non* restartable, but it's
+        # required to expose the exceptions raised by the proxy to the caller.
         for channel_state in channels_to_close:
-            channel_close = ActionChannelClose(
-                token_network_identifier=token_network_identifier,
-                channel_identifier=channel_state.identifier,
+            balance_proof = channel_state.partner_state.balance_proof
+            # silence mypy: partner's balance proofs should be signed
+            assert balance_proof is None or isinstance(balance_proof, BalanceProofSignedState)
```

Comment:
The above seems to skip the following steps:
https://github.com/raiden-network/raiden/blob/a59e6b52a346e751f7b964f6ffcd04ce535339c9/raiden/transfer/channel.py#L1343-L1348

Reply:
Hummm, good catch. Not, it was not, we need to figure out a different way of sending the transaction, this cannot be skipped otherwise the channel will accept transfers while the channel is being closed.